pkg/network/usm/ebpf_main.go

```diff
@@ -222,11 +245,38 @@ func (e *ebpfProgram) Start() error {
 		e.mapCleaner = mapCleaner
 	}
 
+	// Deleting from an array while iterating it is not a simple task. Instead, every successfully enabled protocol,
+	// we'll keep it in a temporary copy, and in case of a mismatch (a.k.a., we have a failed protocols) between
+	// enabledProtocolsTmp to m.enabledProtocols, we'll use the enabledProtocolsTmp.
+	e.enabledProtocols = runForProtocol(e.enabledProtocols, e.Manager.Manager, "pre-start",
```

Comment:
maybe we can find a more indicative name for this function? something like `procesProtocolor`, `excuteProtocol`, I do find the name a bit confusing